src/DeliveryCartReferencedContent.php

```diff
@@ -0,0 +1,189 @@
+
+    if (!empty($routes)) {
+      foreach ($routes as $menuLink) {
+        $plugin = $menuLink->getPluginDefinition();
+        if (isset($plugin) && isset($plugin['metadata']) && isset($plugin['metadata']['entity_id'])) {
+          $id = $plugin['metadata']['entity_id'];
+          $linkEntity = MenuLinkContent::load($id);
+          if ($linkEntity) {
+            \Drupal::service('delivery.cart')->addToCart($linkEntity);
+            self::$count++;
+            self::addMenuParents($linkEntity);
+          }
+        }
+      }
+    }
+
+    return self::$count;
+  }
+
+  /**
+   * Looks for menu parents to also add them to the cart.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   */
+  public static function addMenuParents(EntityInterface $entity) {
+    $parents = $entity->get('parent')->getValue();
+    if (!empty($parents)) {
+      foreach ($parents as $parent) {
+        $parentId = $parent['value'] ? $parent['value'] : '';
+        $parentId = str_replace('menu_link_content:', '', $parentId);
+        $entity = \Drupal::service('entity.repository')
+          ->loadEntityByUuid('menu_link_content', $parentId);
+        if ($entity) {
+          \Drupal::service('delivery.cart')->addToCart($entity);
+          self::$count++;
+          self::addMenuParents($entity);
+        }
+      }
+    }
+  }
+
+  /**
+   * Invokes the reference content hook.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   */
+  public static function referenceContentHook(EntityInterface $entity) {
+    \Drupal::moduleHandler()
+      ->invokeAll('delivery_cart_referenced_content', [$entity]);
+  }
+
+  /**
+   * Looks for any referenced blocks in layout builder and adds them to the
+   * cart.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *
+   * @return false|int
+   */
+  public static function addBlocksFromLayoutBuilder(EntityInterface $entity) {
+    if (!self::hasField($entity, 'layout_builder__layout')) {
+      return FALSE;
+    }
+
+    $sections = $entity->get('layout_builder__layout')->getSections();
+
+    foreach ($sections as $section) {
+      $components = $section->getComponents();
+      foreach ($components as $component) {
+        $renderedArray = $component->toRenderArray();
+        $theme = $renderedArray['#theme'] ? $renderedArray['#theme'] : '';
+        if ($theme === "block") {
+          $content = $renderedArray['content'] ? $renderedArray['content'] : '';
+          if (isset($content['#block_content'])) {
+            $blockContent = $content['#block_content'];
+            \Drupal::service('delivery.cart')->addToCart($blockContent);
+            self::$count++;
+          }
+        }
+      }
+    }
+
+    return self::$count;
+  }
+
+  /**
+   * Adds referenced nodes.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   */
+  public static function addReferencedNodes(EntityInterface $entity) {
+    if ($entity->getEntityTypeId() === "node") {
```

Comment:
Hi @elistone !
Let's remove the entity type condition, and also rename the method to 'addReferencedEntities'.

Reply:
Removed and renamed method to `addReferencedEntities`